### PR TITLE
fix: added fn configureDefaults() to up-ga.js

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

Default GA configuration is not being set. This change adds setting the defaults as global values.

Use of archaic JS compressor prevents use of modern ES6. Will be addressed in uPortal v6, but for now the code was rewritten in an older style that matches the rest of the file.

https://developers.google.com/analytics/devguides/collection/ga4/reference/config#:~:text=Set%20to%20'auto'%20(the,example.com%20for%20the%20domain.

[individual contributor license agreement]: https://github.com/uPortal-Project/uPortal/blob/master/docs/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/uPortal-Project/uPortal/blob/master/docs/CONTRIBUTING.md#commit
[message properties]: https://github.com/uPortal-Project/uPortal/tree/master/uPortal-webapp/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js [hunk 1]
```diff
@@ -45,6 +45,21 @@ var uportal = uportal || {};
         return up.analytics.model.defaultConfig;
     };
 
+    /**
+     * Set the defaultConfig.config array as global settings
+     */
+    var configureDefaults = function (propertyConfig) {
+        //console.log(propertyConfig);
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js:52]: Do we still need the commented out logs?
Could we remove them?

**Review by ChristianMurphy — APPROVED** [review]
LGTM, thanks @bjagg!

File: uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js (hunk 1 above)
Comment (bjagg) [reply uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js:52]: Great question and one I anticipated. I think some of the clients will need to leverage the console logs to conform their configuration is correct. I can go either way. What do you think?

File: uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js [hunk 2]
```diff
@@ -45,6 +45,21 @@ var uportal = uportal || {};
...
+        var defaults = propertyConfig.config || [];
+        _.each(defaults, function (setting) {
+            // each setting is an object that might have more than 1 key-value pair
+            //console.log(setting);
+            _.each(Object.keys(setting), function (key) {
+                up.gtag('set', key, setting[key]);
+            });
+        });
```
[the remark this answers is not in the document]
Comment (bjagg) [reply uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js:60]: lol! That was my plan. Turns out that some of the tooling used does not support ES6. I had to rewrite code back to this.

Thanks for the code. I'll see what works when I have time. Goal is to finish this to release the "final" uPortal v5. Starting on v6 and modernizing tooling like this that holds us back.

[the remark this answers is not in the document]
Comment (ChristianMurphy) [reply uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js:60]: All of what I posted should be ES5 and pass the tooling, unless the compressor doesn't fully support ES5 either 😅

File: uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js
Comment (mhuffsti) [inline uPortal-webapp/src/main/webapp/media/skins/common/javascript/uportal/up-ga.js:1]: @bjagg I took a quick look.  I think @ChristianMurphy hit the programming items that I would have mentioned.  Normally, I try and pull down the code and test it to see if functionally it is doing what is expected.  I have **_not_** done that.  Would you like me to test it out in a development instance as well?

**Comment by mhuffsti** [comment]
@bjagg I went ahead and did a quick functional test.  I focused on cookie_domain which is what started all of this, and it worked as expected this time with the updated JS file.

**Comment by jgribonvald** [comment]
Hi folks, just my 2 cents from european side. Our Protection Data Officers doesn't want to see any Google analytics scripts loaded like initialized even if the configuration is empty. It would be great if this part would be inside an optional stuff, like a jsp portlet code that we won't load by removing the portlet def. I have the same problem with using some CDN links like for google fonts. Google has trackers on.
Analytics are usefull, but it should be managed at a custom way, with an organization platform. So that we host the apps that collect datas and that we provide anonymous stats. We can find alternatives to GA without problems 😉

**Comment by ChristianMurphy** [comment]
> Analytics are usefull, but it should be managed at a custom way, with an organization platform. So that we host the apps that collect datas and that we provide anonymous stats. We can find alternatives to GA without problems 😉

Related, I've had pretty good experiences with Matomo https://matomo.org/
Which is both [self-host-able](https://matomo.org/guide/installation-maintenance/matomo-on-premise-self-hosted/) and [GDPR compliant](https://matomo.org/gdpr-analytics/).

**Comment by jgribonvald** [comment]
> > Analytics are usefull, but it should be managed at a custom way, with an organization platform. So that we host the apps that collect datas and that we provide anonymous stats. We can find alternatives to GA without problems 😉
> 
> Related, I've had pretty good experiences with Matomo https://matomo.org/
> Which is both [self-host-able](https://matomo.org/guide/installation-maintenance/matomo-on-premise-self-hosted/) and [GDPR compliant](https://matomo.org/gdpr-analytics/).

Matomo is largely deployed in France ;)

**Comment by bjagg** [comment]
This PR is focused on a configuration omission. Thanks for the code, @ChristianMurphy.

I'll open an issue about making GA optional and having other solutions.

You guys are awesome!

**Comment by bjagg** [comment]
Yes, looks like the linter and the compressor conflict. Going to revert to the previous version and capture this suggestion as a future issue once we upgrade the JS tooling.

**Pull request merged**